aten/src/ATen/native/native_functions.yaml

```diff
@@ -3357,6 +3357,34 @@
 
 - func: norm.names_out(Tensor self, Scalar? p, Dimname[1] dim, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
 
+- func: _norm_matrix.ScalarOpt_dtype(Tensor self, Scalar? p, *, ScalarType dtype) -> Tensor
+  variants: function
+
+- func: _norm_matrix.Scalar(Tensor self, Scalar p=2) -> Tensor
+  use_c10_dispatcher: full
+  variants: function
+
+- func: _norm_matrix.ScalarOpt_dim_dtype(Tensor self, Scalar? p, int[1] dim, bool keepdim, *, ScalarType dtype) -> Tensor
+  variants: function
+
+- func: _norm_matrix.ScalarOpt_dim(Tensor self, Scalar? p, int[1] dim, bool keepdim=False) -> Tensor
+  use_c10_dispatcher: full
+  variants: function
+
+- func: _norm_matrix.dtype_out(Tensor self, Scalar? p, int[1] dim, bool keepdim, *, ScalarType dtype, Tensor(a!) out) -> Tensor(a!)
+
+- func: _norm_matrix.out(Tensor self, Scalar? p, int[1] dim, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
+
+- func: _norm_matrix.names_ScalarOpt_dim_dtype(Tensor self, Scalar? p, Dimname[1] dim, bool keepdim, *, ScalarType dtype) -> Tensor
```

Comment:
Should this leave named tensors as a future PR for simplicity?

Reply:
Sounds like a good idea, I removed them.